tensorboard/webapp/hparams/_redux/hparams_effects_test.ts

```diff
@@ -0,0 +1,206 @@
+import {buildHparamSpec, buildMetricSpec} from './testing';
+import {HparamAndMetricSpec, RunStatus, SessionGroup} from '../types';
+import * as selectors from '../../selectors';
+import * as runsActions from '../../runs/actions';
+import * as hparamsActions from './hparams_actions';
+import * as appRoutingActions from '../../app_routing/actions';
+import * as coreActions from '../../core/actions';
+import {RouteKind} from '../../app_routing/types';
+
+function mockHparamsDataSourceFactory() {
+  return jasmine.createSpyObj<HparamsDataSource>('HparamsDataSource', [
+    'fetchExperimentInfo',
+    'fetchSessionGroups',
+  ]);
+}
+
+describe('hparams effects', () => {
+  let dataSource: jasmine.SpyObj<HparamsDataSource>;
+  let store: MockStore<State>;
+  let effects: HparamsEffects;
+  let action: ReplaySubject<Action>;
+  let dispatchSpy: jasmine.Spy;
+  let actualActions: Action[];
+
+  let mockHparamsAndMetricsSpecs: HparamAndMetricSpec;
+  let mockSessionGroups: SessionGroup[];
+
+  beforeEach(async () => {
+    action = new ReplaySubject<Action>(1);
+
+    await TestBed.configureTestingModule({
+      providers: [
+        provideMockActions(action),
+        HparamsEffects,
+        provideMockTbStore(),
+        {provide: HparamsDataSource, useFactory: mockHparamsDataSourceFactory},
+      ],
+    }).compileComponents();
+
+    store = TestBed.inject<Store<State>>(Store) as MockStore<State>;
+
+    actualActions = [];
+    dispatchSpy = spyOn(store, 'dispatch').and.callFake((action: Action) => {
+      actualActions.push(action);
+    });
+
+    effects = TestBed.inject(HparamsEffects);
+    dataSource = TestBed.inject(
+      HparamsDataSource
+    ) as jasmine.SpyObj<HparamsDataSource>;
+
+    mockHparamsAndMetricsSpecs = {
+      hparams: [buildHparamSpec({name: 'h1'})],
+      metrics: [buildMetricSpec({tag: 'm1'})],
+    };
+    mockSessionGroups = [
+      {
+        name: 'session_group_1',
+        hparams: {
+          hparam1: 1,
+          hparam2: 'abc',
+        },
+        sessions: [
+          {
+            name: 'exp1/run1',
+            status: RunStatus.STATUS_UNKNOWN,
+            metricValues: [] as any,
+            modelUri: '',
+            monitorUrl: '',
+            startTimeSecs: 123,
+            endTimeSecs: 456,
+          },
+        ],
+      },
+    ];
+
+    dataSource.fetchExperimentInfo.and.returnValue(
+      of(mockHparamsAndMetricsSpecs)
+    );
+
+    dataSource.fetchSessionGroups.and.returnValue(of(mockSessionGroups));
+  });
+
+  describe('loadHparamsData$', () => {
+    beforeEach(() => {
+      effects.loadHparamsData$.subscribe((action) => {
+        actualActions.push(action);
+      });
+      store.overrideSelector(selectors.getEnableHparamsInTimeSeries, true);
+      store.overrideSelector(selectors.getActiveRoute, {
+        routeKind: RouteKind.EXPERIMENT,
+        params: {},
+      });
+      store.overrideSelector(selectors.getExperimentIdsFromRoute, ['exp1']);
```

Comment:
Should we rename to perhaps 'expFromRoute' to make it a bit clearer where the navigated, reload, and manualReload tests are getting this from?